src/ModelContextProtocol.Core/Protocol/Result.cs

```diff
@@ -0,0 +1,19 @@
+using System.Text.Json.Nodes;
+using System.Text.Json.Serialization;
+
+namespace ModelContextProtocol.Protocol;
+
+/// <summary>
+/// Provides a base class for result payloads.
+/// </summary>
+public abstract class Result
```

Comment:
Do we want to expose a common base type if it's not polymorphic?

Reply:
I'm following what was already being done on the request side with request params, and also how it's modeled in the spec in the TypeScript.